site/ruby_scripts/tool.rb

```diff
@@ -0,0 +1,107 @@
+# encoding: utf-8
+
+
+module JekyllPlanet
+
+  def self.run( args )
+    Tool.new.run( args )
+  end
+
+  def self.main
+    self.run( ARGV )
+  end
+
+class Tool
+
+  def initialize()
+    puts "db settings:"
+    @db_config = {
+      adapter: 'sqlite3',
+      database: './planet.db'
+    }
+
+    pp @db_config
+  end
+
+
+  def run( args )
+    unless File.exists?( @db_config[:database])
+      puts "** error: database #{@db_config[:database]} missing; please check pluto documention for importing feeds etc."
+      exit 1;
+    end
+
+    Pluto.connect( @db_config )
+
+    Pluto::Model::Item.latest.limit(10).each_with_index do |item,i|
+      puts "[#{i+1}] #{item.title}"
+  
+      generate_blog_post( item )
+    end
+  end
+
+
+  def generate_blog_post( item )
+
+    posts_root = "./_posts"
+
+    FileUtils.mkdir_p( posts_root )  ## make sure path exists
+
+    ## Note:
+    ## Jekyll pattern for blogs must follow
+    ##  2014-12-21-  e.g. must include trailing dash (-)
+    fn = "#{posts_root}/#{item.published.strftime('%Y-%m-%d')}-#{title_to_key(item.title)}.html"
+
+    frontmatter =<<EOS
+---
+title:      "#{item.title}"
+created_at: #{item.published}
+author:     #{item.feed.title}
+layout:     post
+original_link: "#{item.url}"
+---
+EOS
+
+
+    File.open( fn, 'w' ) do |f|
+      f.write frontmatter
+
+      if item.content
+        f.write item.content
+      elsif item.summary
+        f.write item.summary
+      else
+        ## warn: not content found for feed
+      end
+    end
+  end
+
+
+private
+
+def title_to_key( title )
+  
+  ### fix: use textutils.title_to_key ??
```

Comment:
Please use https://stackoverflow.com/a/7160536/1412255 to replace accented chars with their latin versions instead of the gsub rules below

Reply:
Why is this an issue though ?
Could be a separate one, but this one is generating the url link, which will anyways be url encoded

(Also would be nice to suggest this change upstream at feedreader)

Reply:
This function is used for url right? https://github.com/IIIT-Delhi/byld-website/pull/62/files/c08ad9e5e79520498a5df347411b7f8263b2f10d#diff-38b490977efb3ebb30a78abcd2a6d07dR52

In that case I guess [slugify](https://www.rubydoc.info/gems/textutils/0.9.9/TextUtils/StringFilter#asciify-instance_method) method may be better than defining `title_to_key` at all

Reply:
https://github.com/feedreader/jekyll-planet/blob/master/lib/jekyll/planet/tool.rb

This function was not touched by me, since it worked well

From what I imagine is that the gsub would remove any characters that aren't in the a-z,0-9 set
The rational for gsubbing 
```
  key = key.gsub( 'ü', 'ue' )
  key = key.gsub( 'é', 'e' )
```
seems that it was critical to the original users of this code in order to support 
Probably nice idea to suggest them to switch as well (and they too somewhere had it in mind that's why the fixme)

Reply:
Achha makes sense. I was reviewing this piece of code as if you wrote it.

Reply:
> which will anyways be url encoded

True, but I guessed that the intention of adding this was to make URLs [cleaner](https://stackoverflow.com/questions/47427/why-do-some-websites-add-slugs-to-the-end-of-urls). StackOverflow uses this extensively, and this has become another good practice since such urls are user-friendly. This is why I was suggesting using existing functions to do this instead of implementing custom functionality

> (Also would be nice to suggest this change upstream at feedreader)

True. Though I don't know if they'll change stuff since it's not a general purpose library, they seem to have built it just for their website.

Reply:
Works. This PR could be an "as-is" port and enhancements could be separate issues, no problemo